header-gen/src/cli.ts

```diff
@@ -1,7 +1,9 @@
-import { headerData } from "./inputdata";
-import { generateBlob, generateCArray, generateDsHexLiteral } from "./main";
+import process from 'process';
+import { pathToFileURL } from 'url';
+import { headerData } from "./inputdata.js";
+import { generateBlob, generateCArray, generateDsHexLiteral } from "./main.js";
```

Comment:
@microbit-robert this is the part I haven't had a chance to look into.
As `npm run start` ends up running `node built/cli.js` it looks like the script needs the import to include the extension.

Not sure how to get ts to convert the import statements (in which case maybe we need to reintroduce `tsconfig.cli.json` and npm start would then be `tsc --project tsconfig.cli.json && node built/cli.js`.

Reply:
Assuming the ml extension needs the built import statements as they currently are, then we'd need to make sure the packaging script still used the original `tsconfig.json` and excludes the cli.ts file from the package (as that'd be broken to be used with node and it doesn't do anything as an imported module).